cmd/server.go

```diff
@@ -191,9 +226,47 @@ func setupStorage() (storage.Storage, error) {
 func serveMux() (*http.ServeMux, error) {
 	mux := http.NewServeMux()
 
+	options := []discovery.Option{
+		discovery.WithModulesV1(fmt.Sprintf("%s/", prefixModules)),
+		discovery.WithProvidersV1(fmt.Sprintf("%s/", prefixProviders)),
+	}
+
+	if flagLoginClient != "" {
+		login := &discovery.LoginV1{
+			Client: flagLoginClient,
+		}
+
+		if flagLoginGrantTypes != nil {
+			login.GrantTypes = flagLoginGrantTypes
+		}
+
+		if flagLoginAuthz != "" {
+			login.Authz = flagLoginAuthz
+		}
+
+		if flagLoginToken != "" {
+			login.Token = flagLoginToken
+		}
+
+		if flagLoginPorts != nil {
+			login.Ports = flagLoginPorts
+		}
+
+		if flagLoginScopes != nil {
+			login.Scopes = flagLoginScopes
+		}
```

Comment:
Since we don't have a sensible default for it, we should validate the input values and return an error message for the users